pkg/lorry/plugin/proto/db_plugin.proto

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright (C) 2022-2023 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+syntax = "proto3";
+package plugin.v1;
+
+import "google/protobuf/descriptor.proto";
+
+option go_package = "github.com/apecloud/kubeblocks/pkg/lorry/plugin";
+
+extend google.protobuf.FieldOptions {
+  // Indicates that a field MAY contain information that is sensitive
+  // and MUST be treated as such (e.g. not logged).
+  bool kb_secret = 1059;
+}
+
+message GetPluginInfoRequest {
+  map<string, string> metadata = 1;
+}
+
+message GetPluginInfoResponse {
+  // The name of the plugin. It MUST be 63
+  // characters or less, beginning and ending with an alphanumeric
+  // character ([a-z0-9A-Z]) with dashes (-), dots (.), and
+  // alphanumerics between. This field is REQUIRED.
+  string name = 1;
+
+  string db_type = 2;
+
+  // This field is REQUIRED. 
+  string version = 3;
+
+  // This field is OPTIONAL. 
+  map<string, string> metadata = 4;
+}
+
+message IsDBReadyRequest {
+  map<string, string> metadata = 1;
+}
+
+message IsDBReadyResponse {
+  // true if the DB is ready, false otherwise.
+  bool ready = 1;
+
+  // This field is OPTIONAL. 
+  map<string, string> metadata = 2;
+}
+
+message DBInfo {
+  // The pod FQDN of the replica.
+  string fqdn = 1;
+
+  // The port on which the DB service listens.
+  string port = 2;
+
+  // The admin username used to access the DB service.
+  string admin_user = 3;
+
+  // The admin password of the user used to access the DB service.
+  string admin_password = 4 [(kb_secret) = true];
+}
+
+message GetRoleRequest {
+  DBInfo db_info = 1;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 2;
+}
+
+message GetRoleResponse {
+  string role = 1;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 2;
+}
+
+message JoinMemberRequest {
+  DBInfo db_info = 1;
+
+  // The pod FQDN of the primary replica to check the role.
+  string primary_fqdn = 2;
+
+  // The port on which the primary DB service listens.
+  string primary_port = 3;
+
+  // The Names of all members.
+  repeated string members = 4;
+
+  // The name of the new member's Pod.
+  string new_member = 5;
+
+  // The IP of the new member's Pod.
+  string new_member_ip = 6;
+
```

Comment:
There are three address in this message.
- db_info.fqdn
- primary_fqdn
- new_member

I think only two of them are required, either db_info.fqdn or primary_fqdn.